core/src/main/java/com/xmartlabs/bigbang/core/helper/GeneralErrorHelper.java

```
@@ -17,83 +16,81 @@
 
   @Getter
-  private final io.reactivex.functions.Consumer<? super Throwable> generalErrorAction = t -> {
+  private final Consumer<? super Throwable> generalErrorAction = t -> {
     if (t instanceof CompositeException) {
       CompositeException compositeException = (CompositeException) t;
       Stream.of(compositeException.getExceptions())
           .forEach(this::handleException);
     } else {
       handleException(t);
     }
-    markExceptionAsHandled(t);
   };
 
-  private void handleException(Throwable throwable) {
-    if (!shouldHandleThrowable(throwable)) {
-      if (buildInfo.isDebug()) {
-        Timber.e(throwable, "Untracked exception");
-      } else {
+  private void handleException(@NonNull Throwable throwable) {
+    if (!shouldHandleThrowable(throwable)
+        || (throwable.getCause() != null && !shouldHandleThrowable(throwable.getCause()))) {
+      if (!exceptionIsAlreadyBeingHandled(throwable)) {
         Timber.d(throwable, "Untracked exception");
       }
+      markExceptionAsHandled(throwable);
       return;
     }
 
     if (throwableHandlers.containsKey(throwable.getClass())) {
-      if (exceptionIsAlreadyBeingHandled(throwable)) {
-        return;
-      }
       throwableHandlers.get(throwable.getClass()).accept(throwable);
     } else {
       Timber.e(throwable);
     }
+    markExceptionAsHandled(throwable);
   }
 
-  private boolean shouldHandleThrowable(Throwable throwable) {
+  private boolean shouldHandleThrowable(@NonNull Throwable throwable) {
     return !exceptionIsAlreadyBeingHandled(throwable)
-        || !UNTRACKED_CLASSES.contains(throwable.getClass())
-        || throwableHandlers.containsKey(throwable.getClass());
+        && !UNTRACKED_CLASSES.contains(throwable.getClass());
   }
 
-  private boolean exceptionIsAlreadyBeingHandled(Throwable t) {
-    return Objects.equals(t.getStackTrace(), DUMMY_STACK_TRACE_ELEMENT_ARRAY);
+  private boolean exceptionIsAlreadyBeingHandled(@NonNull Throwable throwable) {
```

isn't it slow for production?

---

we are checking the stacktrace, maybe 100 elements I think we shouldn't have problems. When the exception is not handled all elements will be checked. However, in the other case, the dummy stacktrace element should be at the top of the stacktrace, so I think we shouldn't have performance issues.  
Using the other way we are changing the stacktrace to a dummy stacktrace, I think this solution is better because we keep the original stacktrace (plus a dummy element). make sense?

---

maybe we should change `Objects.equals(value, DUMMY_STACK_TRACE_ELEMENT)` to a memory reference comparison (`value == DUMMY_STACK_TRACE_ELEMENT`).  I think it should be better. What do you think @bryant1410 ?

---

sounds good if it works :smile:

---

yes, the tests are still working :smile: